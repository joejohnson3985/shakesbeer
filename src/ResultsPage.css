.results-page {
  background: white;
  display: block;
  margin: 0;
  color: black;
}

header {
  width: 100%;
  background: #f78f2f;
  display: flex;
  align-items: center;
  border-top: solid 5px #f9a960;
  border-bottom: solid 5px #f9a960;
}

header .logo {
  height: 100px;
  width: 100px;
  padding-bottom: 10px;
  padding-left: 10px;
  padding-right: 5px;
}

h1.results-header {
  color: rgb(255, 255, 255);
  font-size: 100px;
  margin: 0;
  padding: 0;
}

h2.state-subheading {
  color: rgb(87, 87, 87);
  /* margin: 0px 30px; */
}

main {
  margin: 0;
  padding: 10px 30px;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}



