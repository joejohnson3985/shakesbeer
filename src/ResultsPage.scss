@import './variables.scss';

.results-page {
  background: white;
  display: block;
  margin: 0;
  color: black;
}

header {
  width: 100%;
  background: $primary-color;
  display: flex;
  align-items: center;
  border-top: solid 5px $secondary-color;
  border-bottom: solid 5px $secondary-color;

  .logo {
    height: 80px;
    width: 90px;
    padding-bottom: 10px;
    padding-left: 10px;
    padding-right: 5px;
  }
  
  .results-header {
    color: rgb(255, 255, 255);
    font-size: 100px;
    margin: 0;
    padding: 0;
  }
}

.brewery-list-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0px 15px;
  width: 60%;
}

.flag {
  height: 50px;
  border: 1px solid gray;
}

.state-subheading {
  color: rgb(87, 87, 87);
  font-size: 32px;
  padding: 0px 20px;
}

main {
  margin: 0;
  padding: 10px 30px;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.hidden {
  display: none;
}

.fa-arrow-left {
  font-size: 30px;
  margin: 20px 10px 20px 20px;
  padding: 10px;
  border-radius: 100%;
  cursor: pointer;

  &:hover {
    color: $primary-color;
    background: white;
  }
}

