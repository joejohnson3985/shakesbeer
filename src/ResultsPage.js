import React, { Component } from 'react';
import './ResultsPage.css';
import Controls from './Controls.js'
import BreweryList from './BreweryList.js'
import './images/shakesbeerlogosmall.png';

class ResultsPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      stateBreweries: [],
      filteredBreweries: [],
      breweryCities: [],
      beerStyles: [],
      beerIbus: [],
      citySelection: 'All', 
      styleSelection: 'All',
      ibuSelection: 'All', 
      abvSelection: 'All'
    }
  }

  componentWillMount() {
    this.getStateBreweries();
  }

  getStateBreweries = () => {
    let stateBreweries = this.props.dataset.filter(brewery => {
      return brewery.state === this.props.selectedState;
    })
    this.setState({ stateBreweries: stateBreweries }, () => {
      this.refreshBreweryList();
      this.getCities();
      this.getStyles();
      this.getIbus();
    })
  }

  getCities = () => {
    let cities = this.state.stateBreweries.reduce((acc, brewery) => {
      if(!acc.includes(brewery.city)) {
        acc.push(brewery.city)
      }
      return acc;
    }, []).sort();
    this.setState({ breweryCities: cities });
  } 

  getStyles = () => {
    let styles = this.state.stateBreweries.reduce((acc, brewery) => {
      brewery.beers.forEach(beer => {
        if(!acc.includes(beer.style) && beer.style !== '') {
          acc.push(beer.style)
        }
      })
      return acc;
    }, []).sort();

    this.setState({ beerStyles: styles });
  } 

  getIbus = () => {
    let ibus = this.state.stateBreweries.reduce((acc, brewery) => {
      brewery.beers.forEach(beer => {
        const ibuMinRange = Math.round(beer.ibu / 10) * 10;
        const ibuOption = `${ibuMinRange}-${ibuMinRange + 9}`;
        if(!acc.includes(ibuOption) && beer.ibu !== null) {
          acc.push(ibuOption);
        }
      })
      return acc;
    }, []).sort().sort((a, b) => a.length - b.length);
    this.setState({ beerIbus: ibus });
  }

  updateFilterSelection = (filterName, value) => {
    const propertyName = `${filterName}Selection`;
    this.setState({ [propertyName]: value });
    this.refreshBreweryList();
  }

  refreshBreweryList = () => {
    //reset page defaults, or whatever we need to do to get beer details to show and collapse on filter change
    this.setState({ filteredBreweries: [...this.state.stateBreweries] }, () => this.filterByCity());
  }

  filterByCity = () => {
    if (this.state.citySelection !== 'All') {
      let filterResults = this.state.filteredBreweries.filter(brewery => {
        return brewery.city === this.state.citySelection;
      });
      this.setState({ filteredBreweries: filterResults }, () => this.filterByStyle());
    } else {
      this.filterByStyle()
    }
  }

  filterByStyle = () => {
    if (this.state.styleSelection !== 'All') {
      let filterResults = this.state.filteredBreweries.reduce((acc, brewery) => {
        let breweryStyles = brewery.beers.map(beer => beer.style);
        if(breweryStyles.includes(this.state.styleSelection)) {
          acc.push(brewery);
        }
        return acc;
      }, [])
      this.setState({filteredBreweries: filterResults}, () => this.filterByIbu());
    } else {
      this.filterByIbu();
    }
  }

  filterByIbu = () => {
    if (this.state.ibuSelection !== 'All') {
      let filterResults = this.state.filteredBreweries.reduce((acc, brewery) => {
        let breweryIbus = brewery.beers.map(beer => {
          return `${Math.round(beer.ibu / 10) * 10}-${(Math.round(beer.ibu / 10) * 10) + 9}`;
        });
        if(breweryIbus.includes(this.state.ibuSelection)) {
          acc.push(brewery);
        }
        return acc;
      }, [])
      this.setState({filteredBreweries: filterResults}); //add filter by ABV to callback
    } 
    // else {
    //   this.filterByAbv()
    // }
  }
 
  render() {
    // console.log(this.state.breweryCities)
    return (
      <div className="results-page">
        <header>
          <img src="./images/shakesbeerlogosmall.png" alt="shakesbeer logo"/>
          <h1 className="results-header">ShakesBeer</h1>
        </header>
        <main>
          <h2 className="state-subheading">{this.props.selectedState}</h2>  
          <div className="brew-cards">
            <Controls breweryCities={this.state.breweryCities} 
                      beerStyles={this.state.beerStyles}
                      beerIbus={this.state.beerIbus}
                      updateFilterSelection={this.updateFilterSelection}
            />
            <BreweryList filteredBreweries={this.state.filteredBreweries} 
                         dataset={this.props.dataset}
            />
          </div>
        </main>
      </div>
    );
  }
}

export default ResultsPage;