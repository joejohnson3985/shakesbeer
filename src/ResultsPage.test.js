import React from 'react';
import ResultsPage from './ResultsPage.js';
import { shallow } from 'enzyme';

const mockData = [1,2,3,4]
const mockFunc = jest.fn()

describe('ResultsPage', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
          <ResultsPage selectedState={mockData}
                       dataset={mockData}
          />
      )
  });
  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });





});