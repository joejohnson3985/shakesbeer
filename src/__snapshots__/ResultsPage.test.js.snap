// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage Should match the snapshot 1`] = `
<div
  className="results-page"
>
  <header>
    <img
      alt="shakesbeer logo"
      src="./images/shakesbeerlogosmall.png"
    />
    <h1
      className="results-header"
    >
      ShakesBeer
    </h1>
  </header>
  <main>
    <h2
      className="state-subheading"
    >
      <Component />
    </h2>
    <div
      className="brew-cards"
    >
      <Controls
        beerIbus={Array []}
        beerStyles={Array []}
        breweryCities={Array []}
        updateFilterSelection={[Function]}
      />
      <BreweryList
        dataset={
          Array [
            Object {
              "FIELD1": 90,
              "beers": Array [
                Object {
                  "FIELD1": 43,
                  "abv": 0.045,
                  "brewery_id": 60,
                  "ibu": 18,
                  "id": 2544,
                  "name": "Send Help",
                  "ounces": 12,
                  "style": "American Blonde Ale",
                },
                Object {
                  "FIELD1": 44,
                  "abv": 0.055,
                  "brewery_id": 60,
                  "ibu": null,
                  "id": 2324,
                  "name": "Cast Iron Oatmeal Brown",
                  "ounces": 12,
                  "style": "American Brown Ale",
                },
                Object {
                  "FIELD1": 45,
                  "abv": 0.06,
                  "brewery_id": 60,
                  "ibu": null,
                  "id": 2288,
                  "name": "Reprise Centennial Red",
                  "ounces": 12,
                  "style": "American Amber / Red Ale",
                },
                Object {
                  "FIELD1": 46,
                  "abv": 0.055,
                  "brewery_id": 60,
                  "ibu": null,
                  "id": 2287,
                  "name": "Alter Ego",
                  "ounces": 12,
                  "style": "American Black Ale",
                },
              ],
              "city": "Wilkes-Barre",
              "name": "The Lion Brewery",
              "state": "Pennsylvania",
            },
          ]
        }
        filteredBreweries={Array []}
      />
    </div>
  </main>
</div>
`;
